### PR TITLE
Don't attempt to make file non-executable under Windows

### LINKED ISSUE #734: Error while saving file to cache on Windows OS (PosixFileAttributeView not supported)
Remove the .lemminx folder cache, create a XSD file and open completion

```xml
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
	targetNamespace="http://ui">
|
<xs:schema>
```

The file is correctly created in the .lemminx folder, but in the log I have this error. This bug occurs on Windows OS .

```
Error - 11:31:50] May 28, 2020 11:31:50 org.eclipse.lemminx.utils.FilesUtils saveToFile()
Message: The associated file system 'C:\Users\azerr\.lemminx\cache\http\www.w3.org\2001\XMLSchema.xsd' + does not support the PosixFileAttributeView
java.lang.UnsupportedOperationException
	at java.base/java.nio.file.Files.setPosixFilePermissions(Files.java:2078)
	at org.eclipse.lemminx.utils.FilesUtils.saveToFile(FilesUtils.java:155)
	at org.eclipse.lemminx.utils.FilesUtils.saveToFile(FilesUtils.java:135)
	at org.eclipse.lemminx.uriresolver.CacheResourcesManager.getResourceCachePath(CacheResourcesManager.java:207)
	at org.eclipse.lemminx.extensions.xsd.XSDURIResolverExtension.resolve(XSDURIResolverExtension.java:61)
	at org.eclipse.lemminx.uriresolver.URIResolverExtensionManager.resolve(URIResolverExtensionManager.java:93)
	at org.eclipse.lemminx.extensions.contentmodel.model.ContentModelManager.findCMDocument(ContentModelManager.java:180)
	at org.eclipse.lemminx.extensions.contentmodel.model.ContentModelManager.findCMDocument(ContentModelManager.java:131)
	at org.eclipse.lemminx.extensions.contentmodel.model.ContentModelManager.findCMDocument(ContentModelManager.java:94)
	at org.eclipse.lemminx.extensions.contentmodel.model.ContentModelManager.findCMDocument(ContentModelManager.java:90)
	at org.eclipse.lemminx.extensions.contentmodel.participants.ContentModelCompletionParticipant.onTagOpen(ContentModelCompletionParticipant.java:71)
	at org.eclipse.lemminx.services.XMLCompletions.collectOpenTagSuggestions(XMLCompletions.java:608)
	at org.eclipse.lemminx.services.XMLCompletions.collectInsideContent(XMLCompletions.java:733)
	at org.eclipse.lemminx.services.XMLCompletions.doComplete(XMLCompletions.java:224)
	at org.eclipse.lemminx.services.XMLLanguageService.doComplete(XMLLanguageService.java:139)
	at org.eclipse.lemminx.XMLTextDocumentService.lambda$completion$1(XMLTextDocumentService.java:189)
	at java.base/java.util.concurrent.CompletableFuture.biApply(CompletableFuture.java:1236)
	at java.base/java.util.concurrent.CompletableFuture$BiApply.tryFire(CompletableFuture.java:1205)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)

```

### PR DESCRIPTION
**Pull request by datho7561**
Fixes #734

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/FilesUtils.java [hunk 1]
```diff
@@ -153,13 +153,16 @@ public static void saveToFile(String content, Path outFile) throws IOException {
...
 		// Make sure it's not executable
-		try {
-			Files.setPosixFilePermissions(outFile, PosixFilePermissions.fromString("rw-r-----"));
-		} catch (UnsupportedOperationException e) {
-			// The associated file system does not support the PosixFileAttributeView,
-			// ignore the error
-			LOGGER.log(Level.SEVERE,
-					"The associated file system '" + outFile + "' + does not support the PosixFileAttributeView", e);
+		// Windows doesn't support marking files as executable/not-executable
+		if (!Platform.isWindows) {
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/FilesUtils.java:157]: I wonder if correct fix should be just:

`outFile.toFile().setExecutable(false);`

which should work for any OS. @fbricon what do you think about that?

Comment (fbricon) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/FilesUtils.java:157]: yeah I guess it should work. Unit tests will confirm ;-)

**Comment by angelozerr** [comment]
@datho7561 I think you can swith your draft PR to real PR, no?

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/utils/FilesUtilsTest.java [hunk 2]
```diff
@@ -59,4 +64,14 @@ public void cleanPathForWindows() {
...
 	}
+
+	@Test
+	public void saveFileAndRemoveExecutableFlag(@TempDir Path temporaryDirectory) throws IOException {
+		Path tempFilePath = temporaryDirectory.resolve("file.xml");
+		FilesUtils.saveToFile("<content />", tempFilePath);
+		File tempFile = tempFilePath.toFile();
+		assertTrue(tempFile.exists());
+		assertTrue(tempFile.isFile());
+		assertTrue(!tempFilePath.toFile().canExecute() || Platform.isWindows);
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/utils/FilesUtilsTest.java:75]: `assertTrue(!tempFile.canExecute() || Platform.isWindows);`

**Comment by angelozerr** [comment]
Thanks @datho7561

**Pull request merged**